examples/with-aptos/src/index.ts

```diff
@@ -0,0 +1,142 @@
+  const aptosPublicKeyHex = process.env.APTOS_PUBLIC_KEY!;
+
+  if (!aptosAddress || !aptosPublicKeyHex) {
+    console.error(
+      "Please set your APTOS_ADDRESS and APTOS_PUBLIC_KEY in the .env.local file."
+    );
+    process.exit(1);
+  }
+
+  console.log(`Using Aptos address: ${aptosAddress}`);
+  const publicKeyBytes = Uint8Array.from(Buffer.from(aptosPublicKeyHex, "hex"));
+
+  // Check if account exists
+  let accountData = await client.getAccount(aptosAddress).catch(() => null);
+  if (!accountData) {
+    console.log(
+      `Your account does not exist. Please fund your address ${aptosAddress} to proceed.`
+    );
+    process.exit(1);
+  }
+
+  // Create and sign a transaction
+  const recipientAddress = await input({
+    message: "Recipient address:",
+    default: "<recipient_aptos_address>",
+  });
+
+  const amount = 100n; // 100 Octas (minimum practical amount)
+
+  console.log(
+    `\nSending ${amount} Octas (${
+      Number(amount) / 1e8
+    } APT) to ${recipientAddress}`
+  );
+
+  // Prepare the transaction payload
+  const token = new TxnBuilderTypes.TypeTagStruct(
+    TxnBuilderTypes.StructTag.fromString("0x1::aptos_coin::AptosCoin")
+  );
+
+  const entryFunctionPayload =
+    new TxnBuilderTypes.TransactionPayloadEntryFunction(
+      TxnBuilderTypes.EntryFunction.natural(
+        "0x1::coin",
+        "transfer",
+        [token],
+        [
+          BCS.bcsToBytes(
+            TxnBuilderTypes.AccountAddress.fromHex(recipientAddress)
+          ),
+          BCS.bcsSerializeUint64(Number(amount)),
+        ]
+      )
+    );
+
+  // Get account sequence number and chain ID
+  const [{ sequence_number: sequenceNumber }, chainId] = await Promise.all([
+    client.getAccount(aptosAddress),
+    client.getChainId(),
+  ]);
+
+  // Build the raw transaction
+  const rawTxn = new TxnBuilderTypes.RawTransaction(
+    TxnBuilderTypes.AccountAddress.fromHex(aptosAddress),
+    BigInt(sequenceNumber),
+    entryFunctionPayload,
+    2000n, // Max gas amount
+    100n, // Gas price per unit
+    BigInt(Math.floor(Date.now() / 1000) + 600), // Expiration timestamp
+    new TxnBuilderTypes.ChainId(Number(chainId))
+  );
+
+  // Get the signing message
+  const signingMessage = TransactionBuilder.getSigningMessage(rawTxn);
+
+  // Sign the transaction using Turnkey with HASH_FUNCTION_NO_OP
+  const txSignResult = await turnkeyClient.apiClient().signRawPayload({
+    signWith: aptosAddress,
+    payload: bytesToHex(signingMessage),
+    encoding: "PAYLOAD_ENCODING_HEXADECIMAL",
+    hashFunction: "HASH_FUNCTION_NOT_APPLICABLE",
+  });
+
+  // Extract r and s from the result
+  const { r, s } = txSignResult;
+
+  // Ensure r and s are 64 hex characters (32 bytes)
+  const rHex = r.padStart(64, "0");
+  const sHex = s.padStart(64, "0");
+
+  // Concatenate r and s to form the signature
+  const txSignatureHex = rHex + sHex;
+
+  // Validate signature length
+  if (txSignatureHex.length !== 128) {
+    console.error(
```

Comment:
we could just throw an error here

```
throw new Error("Invalid signature length for Ed25519. Expected 128 hex characters.") 
```

and the handler (`main().catch()...`) will catch the error and exit